keanu-project/src/test/java/io/improbable/keanu/vertices/VertexMatchers.java

```diff
@@ -60,6 +61,10 @@ public void describeTo(Description description) {
         return hasValue(Arrays.stream(values).map(v -> equalTo(v)).collect(Collectors.toList()));
     }
 
+    public static <DATATYPE, TENSOR extends Tensor<DATATYPE>> Matcher<Vertex<TENSOR>> hasTensorValue(TENSOR tensor) {
```

Comment:
Similar comment as for `TensorMatchers` - if we override `Tensor#equals` then your tests would work with `hasValue` and you wouldn't need `hasTensorValue`.